### PR TITLE
CPack Updates

### LINKED ISSUE #1522: CI `save-state` and `set-output` commands are deprecated.
See this recent action output - https://github.com/longturn/freeciv21/actions/runs/3518390686

https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/

### LINKED ISSUE #1525: Generate DMG package with Cpack
**Is your feature request related to a problem? Please describe.**
Right now the macOS CI will build a `dmg` file with custom code. Look to standardize on `cpack`

**Describe the solution you'd like**
Use this generator instead of custom code: https://cmake.org/cmake/help/v3.22/cpack_gen/dmg.html

**Describe alternatives you've considered**
None

**Additional context**
None

### LINKED ISSUE #1611: Installers don't remove deleted files
**Describe the bug**
If we remove files from the repo between releases, the binary installers do not remove them at install time.

**To Reproduce**
1. Install beta.6 (any os)
2. Install beta.7 over beta.6 (any os)
3. notice both libertine and libertinus fonts are still there. (we got rid of libertine for libtertinus between releases).

**Expected behavior**
Installers clean up dead files. The reproduce steps is just one example. We have changed icons/images in the themes and removed files that won't get removed. In the release after beta.7 we are going to get rid of dead themes that won't get removed as intended.

**Screenshots**
n/a

**Platform and version (please complete the following information):**
 - OS: Linux .deb, macOS .dmg and Windows .exe
 - Freeciv21 version: 3.0-beta.7
 - Ruleset/Longturn game (if applicable): n/a

**Additional context**

### PR DESCRIPTION
**Pull request by jwrober**
- Use the git tag in the file name for packages (get rid of the integer value for patch) to match other archives
- Add macOS package
- Get windows nsis installer to uninstall previous version

Closes #1611 
Closes #1525 
Closes #1522

### CODE REVIEW

File: cmake/CPackConfig.cmake [hunk 1]
```diff
@@ -111,5 +114,23 @@ if(UNIX AND NOT APPLE)
...
+
+  set(CPACK_BINARY_BUNDLE "ON")
+  set(CPACK_BINARY_DRAGNDROP "ON")
+  set(CPACK_PACKAGE_DIRECTORY "${CMAKE_BINARY_DIR}")
+  set(CPACK_OUTPUT_FILE_PREFIX ${CPACK_PACKAGE_DIRECTORY})
+  set(CPACK_CPU_ARCH ${CMAKE_SYSTEM_PROCESSOR})
+  set(CPACK_SYSTEM_NAME "${CMAKE_SYSTEM_NAME}-${CPACK_CPU_ARCH}")
+  set(CPACK_PACKAGE_FILE_NAME "${CPACK_PACKAGE_NAME}-${FC21_REV_TAG}-${CPACK_SYSTEM_NAME}")
+  set(CPACK_COMPONENTS_GROUPING ALL_COMPONENTS_IN_ONE)
+  set(CPACK_BUNDLE_NAME "Freeciv21 Installer")
```
Comment (lmoureaux) [inline cmake/CPackConfig.cmake:128]: I don't think it's an *Installer*, it's just Freeciv21.

File: cmake/CPackConfig.cmake [hunk 2]
```diff
@@ -111,5 +114,23 @@ if(UNIX AND NOT APPLE)
...
+  set(CPACK_BUNDLE_PLIST "${CPACK_SOURCE_DIR}/dist/Info.plist")
+  set(CPACK_BUNDLE_ICON "${CPACK_SOURCE_DIR}/dist/client.ico")
+  set(CPACK_DMG_VOLUME_NAME "Freeciv21 Installer")
```
Comment (lmoureaux) [inline cmake/CPackConfig.cmake:131]: Same here, I think it's just Freeciv21.

**Review by lmoureaux — COMMENTED** [review]
I'm not sure how the naming changes affect the rest of the CI, maybe we need to change artifact names somewhere.
It would be nice to have an explanation in the commit messages for future reference, right now they look a bit random.

File: cmake/CPackConfig.cmake (hunk 1 above)
Comment (jwrober) [reply cmake/CPackConfig.cmake:128]: Ok let me get it working and then can fine tune. Using CI as a test bed since I don't have a mac.

**Comment by jwrober** [comment]
> I'm not sure how the naming changes affect the rest of the CI, maybe we need to change artifact names somewhere. It would be nice to have an explanation in the commit messages for future reference, right now they look a bit random.

Not sure I am following what you mean here...

**Comment by lmoureaux** [comment]
> Not sure I am following what you mean here...

You changed some variables with `NAME` in them, I assume they refer to file names?

**Comment by jwrober** [comment]
OIC. No real impact to the CI. Its just the CPack output filename for the package.

**Comment by lmoureaux** [comment]
For instance, will this one work if the file name changes:
```sh
shasum -a 256 Freeciv21-${{steps.split.outputs.fragment}}.dmg > Freeciv21-${{steps.split.outputs.fragment}}.dmg.sha256
```
Seems yes since the CI passes :smile:

**Pull request closed without merging**